src/Renci.SshNet/Abstractions/SocketAbstraction.cs

```diff
@@ -129,7 +129,7 @@ public static void ClearReadBuffer(Socket socket)
 
         public static int ReadPartial(Socket socket, byte[] buffer, int offset, int size, TimeSpan timeout)
         {
-            socket.ReceiveTimeout = (int) timeout.TotalMilliseconds;
+            socket.ReceiveTimeout = timeout.AsTimeout(nameof(socket));
```

Comment:
Why `nameof(socket)`? not `nameof(timeout)`?

Reply:
Fixed.